enterprise/internal/scim/user.go

```diff
@@ -39,16 +41,117 @@ func NewUserResourceHandler(ctx context.Context, observationCtx *observation.Con
 }
 
 // Create stores given attributes. Returns a resource with the attributes that are stored and a (new) unique identifier.
-func (h *UserResourceHandler) Create(r *http.Request, attributes scim.ResourceAttributes) (scim.Resource, error) {
-	// TODO: For testing only, real logic should come here
-	attributesString := resourceAttributesToLoggableString(attributes)
-	h.observationCtx.Logger.Error("XXXXX Create", log.String("method", r.Method), log.String("attributes", attributesString))
+func (h *UserResourceHandler) Create(_ *http.Request, attributes scim.ResourceAttributes) (scim.Resource, error) {
+	// Get external ID
+	optionalExternalID := getOptionalExternalID(attributes)
+
+	// Get primary email: try to get the (first) email address marked as primary, otherwise use the first email address
+	primaryEmail := ""
+	if attributes["emails"] == nil {
+		return scim.Resource{}, scimerrors.ScimErrorBadParams([]string{"emails missing"})
+	}
+	emails := attributes["emails"].([]interface{})
+	for _, email := range emails {
+		if email.(map[string]interface{})["primary"] == true {
+			primaryEmail = email.(map[string]interface{})["value"].(string)
```

Comment:
```suggestion
	emails := attributes["emails"].([]map[string]interface{})
	for _, email := range emails {
		if email["primary"] == true {
			primaryEmail = email["value"].(string)
```

I think this should work, but I do agree with @kopancek further down: not a huge fan of the casting here. since you use an external type, there might not be much we can do.

Reply:
Thanks! I've tried this today after reading @kopancek's suggestions and it didn't work. The most I could do was [this](https://github.com/sourcegraph/sourcegraph/pull/47573/commits/4786c513f6ee7070f8a93c576da23eeb8a68cc5b), which made the code somewhat cleaner.